op-service/txmgr/txmgr_test.go

```diff
@@ -1465,6 +1470,53 @@ func TestMinFees(t *testing.T) {
 	}
 }
 
+func TestMaxFees(t *testing.T) {
+	for _, tt := range []struct {
+		desc             string
+		maxBaseFee       *big.Int
+		maxTipCap        *big.Int
+		expectMaxBaseFee bool
+		expectMaxTipCap  bool
+	}{
+		{
+			desc: "no-maxs",
+		},
+		{
+			desc:             "max-basefee",
+			maxBaseFee:       big.NewInt(baseBaseFee - 1),
+			expectMaxBaseFee: true,
+		},
+		{
+			desc:            "max-tipcap",
+			maxTipCap:       big.NewInt(baseGasTipFee - 1),
+			expectMaxTipCap: true,
+		},
+	} {
+		t.Run(tt.desc, func(t *testing.T) {
+			require := require.New(t)
+			conf := configWithNumConfs(1)
+			conf.MaxBaseFee.Store(tt.maxBaseFee)
+			conf.MaxTipCap.Store(tt.maxTipCap)
+			h := newTestHarnessWithConfig(t, conf)
+
+			tip, baseFee, _, err := h.mgr.SuggestGasPriceCaps(context.TODO())
```

Comment:
Would it be better to pass in a background context or even the test context?

Reply:
Done in 12da053f2b3e895fea804e805deadab3df28fb60.